screens/StartScreen.tsx

```diff
@@ -1,98 +1,138 @@
-import React, { useEffect, useState } from "react";
+import React, { useCallback, useEffect, useState } from "react";
 import { Image, ScrollView, StyleSheet, useColorScheme } from "react-native";
 import { Text, View } from "../components/Themed";
-import { getAllDocsInCollection } from "../helper";
+import { getAllDocsInCollection, getOneDocById } from "../helper";
 import { Review } from "../Interfaces";
 import Tabbar from "../components/Tabbar";
 import { RootTabScreenProps } from "../types";
 import { ActivityCard } from "../components/ActivityCard";
 import Colors, { gradientDark, gradientLight } from "../constants/Colors";
 import { LinearGradient } from "expo-linear-gradient";
 import { useIsFocused } from "@react-navigation/native";
+import { useSelector } from "react-redux";
+import { currentReduxUser } from "../redux/signin";
+import LoadingSpinner from "../components/LoadingSpinner";
 
 export default function StartScreen({
   navigation,
 }: RootTabScreenProps<"Home">) {
+  const myUser = useSelector(currentReduxUser);
   const [reviews, setReviews] = useState<Review[]>([]);
+  const myFollowingArray: string[] = [];
   const colorScheme: any = useColorScheme();
   let isLight = colorScheme == "light" ? true : false;
   const isFocused = useIsFocused();
 
   useEffect(() => {
     if (isFocused) {
-      getReviews();
+      getMyFollowing();
     }
   }, [isFocused]);
 
+  const getMyFollowing = async () => {
+    try {
+      const user = await getOneDocById("users", myUser.id);
+      if (user?.following) {
+        myFollowingArray.push(...user.following);
+      }
+      getReviews();
+    } catch (err) {
+      console.log(err);
+    }
+  };
+
   const getReviews = async () => {
+    getMyFollowing();
     let newData: Review[] = [];
     try {
       let data = await getAllDocsInCollection("recensioner");
```

Comment:
It would be better if you only fetched the specific reviews based on users id and not all of them. That will just take longer time and degrades performance.

I would recommend using getDocsWithSpecificValue instead. Then you dont need the sorting function either.